File: src/main/vrml/net/coderextreme/x3d_code/www_web3d_org/x3d/content/examples/Basic/course/SimpleExtrusionShipHull.js
let browser = X3D.getBrowser();
let X3D0 = {};
X3D0.profile = "Immersive";
X3D0.version = "3.0";
let WorldInfo2 = browser.currentScene.createNode("WorldInfo");
WorldInfo2.title = "SimpleExtrusionShipHull.x3d";
browser.currentScene.children = new MFNode();

browser.currentScene.children[0] = WorldInfo2;

let Viewpoint3 = browser.currentScene.createNode("Viewpoint");
Viewpoint3.description = "Extruded ship hull and deck";
Viewpoint3.orientation = new SFRotation(new float[1,0,0,-0.2]);
Viewpoint3.position = new SFVec3f(new float[-2,8,40]);
browser.currentScene.children[1] = Viewpoint3;

let Transform4 = browser.currentScene.createNode("Transform");
let Shape5 = browser.currentScene.createNode("Shape");
let Extrusion6 = browser.currentScene.createNode("Extrusion");
Extrusion6.creaseAngle = 1.57;
Extrusion6.crossSection = new MFVec2f(new float[0,1,0.38,0.92,0.71,0.71,0.92,0.38,1,0,0.92,-0.38,0.71,-0.71,0.38,-0.92,0,-1]);
Extrusion6.orientation = new MFRotation(new float[0,1,0,0,0,1,0,0,0,1,0,0,0,0,1,-0.5]);
Extrusion6.scale = new MFVec2f(new float[4,4,4,4,4,4,4,0.1]);
Extrusion6.solid = False;
Extrusion6.spine = new MFVec3f(new float[-20,0,0,0,0,0,10,0,0,18,0.25,0]);
Shape5.geometry = Extrusion6;

let Appearance7 = browser.currentScene.createNode("Appearance");
let Material8 = browser.currentScene.createNode("Material");
Material8.diffuseColor = new SFColor(new float[0,0.3,1]);
Material8.emissiveColor = new SFColor(new float[0.2,0.2,0.2]);
Appearance7.material = Material8;

Shape5.appearance = Appearance7;

Transform4.children = new MFNode();

Transform4.children[0] = Shape5;

let Shape9 = browser.currentScene.createNode("Shape");
let Extrusion10 = browser.currentScene.createNode("Extrusion");
Extrusion10.creaseAngle = 1.57;
Extrusion10.crossSection = new MFVec2f(new float[0.1,0.96,0.1,-0.96]);
Extrusion10.scale = new MFVec2f(new float[4,4,4,4,4,4,4,0.1]);
Extrusion10.solid = False;
Extrusion10.spine = new MFVec3f(new float[-19.9,0,0,0,0,0,10,0,0,17.5,0.25,0]);
Shape9.geometry = Extrusion10;

let Appearance11 = browser.currentScene.createNode("Appearance");
let Material12 = browser.currentScene.createNode("Material");
Material12.diffuseColor = new SFColor(new float[0,1,0]);
Material12.emissiveColor = new SFColor(new float[0,0.2,0]);
Appearance11.material = Material12;

Shape9.appearance = Appearance11;

Transform4.children[1] = Shape9;

browser.currentScene.children[2] = Transform4;

File: src/main/python/net/x3djsonld/data/c/x3d-code/www.web3d.org/x3d/content/examples/Basic/course/SimpleExtrusionShipHull.py
###############################################
#
# Now available: developmental python x3d.py package on PyPi for import.
#   This approach greatly simplifies Python X3D deployment and use.
#   https://pypi.org/project/x3d
#
# Installation:
#       pip install x3d
# or
#       python -m pip install x3d
#
# Developer options for loading x3d package:
#
#    from x3d import *  # preferred approach, terser source that avoids x3d.* class prefixes
#
# or
#    import x3d         # traditional way to subclass x3d package, all classes require x3d.* prefix
#                       # but python source is very verbose, for example x3d.Material x3d.Shape etc.
#                       # X3dToPython.xslt stylesheet insertPackagePrefix=true supports this option.

from x3d import *

###############################################

newModel=X3D(profile='Immersive',version='3.0',
  head=head(
    children=[
    meta(content='SimpleExtrusionShipHull.x3d',name='title'),
    meta(content='Don Brutzman and MV4204 class',name='creator'),
    meta(content='29 October 2000',name='created'),
    meta(content='20 October 2019',name='modified'),
    meta(content='Using Extrusion to model a ship hull and deck.',name='description'),
    meta(content='https://www.web3d.org/x3d/content/examples/Basic/course/SimpleExtrusionShipHull.x3d',name='identifier'),
    meta(content='X3D-Edit 3.3, https://savage.nps.edu/X3D-Edit',name='generator'),
    meta(content='../license.html',name='license')]),
  Scene=Scene(
    children=[
    WorldInfo(title='SimpleExtrusionShipHull.x3d'),
    Viewpoint(description='Extruded ship hull and deck',orientation=(1,0,0,-0.2),position=(-2,8,40)),
    Transform(
      children=[
      Shape(
        geometry=Extrusion(creaseAngle=1.57,crossSection=[(0.00,1.00),(0.38,0.92),(0.71,0.71),(0.92,0.38),(1.00,0.00),(0.92,-0.38),(0.71,-0.71),(0.38,-0.92),(0.00,-1.00)],orientation=[(0,1,0,0),(0,1,0,0),(0,1,0,0),(0,0,1,-0.5)],scale=[(4,4),(4,4),(4,4),(4,.1)],solid=False,spine=[(-20,0,0),(0,0,0),(10,0,0),(18,0.25,0)]),
        appearance=Appearance(
          material=Material(diffuseColor=(0,0.3,1),emissiveColor=(0.2,0.2,0.2)))),
      Shape(
        geometry=Extrusion(creaseAngle=1.57,crossSection=[(0.10,0.96),(0.10,-0.96)],scale=[(4,4),(4,4),(4,4),(4,.1)],solid=False,spine=[(-19.9,0,0),(0,0,0),(10,0,0),(17.5,0.25,0)]),
        appearance=Appearance(
          material=Material(diffuseColor=(0,1,0),emissiveColor=(0,0.2,0))))])])
) # X3D model complete

###############################################
# Self-test diagnostics
###############################################

if        metaDiagnostics(newModel): # built-in utility method in X3D class
    print(metaDiagnostics(newModel))
print('check  newModel.XML() serialization...')
newModelXML = newModel.XML() # test export method XML() for exceptions
# print(newModelXML) # debug

print ("python x3d.py load successful for SimpleExtrusionShipHull.py")
